include/beman/inplace_vector/inplace_vector.hpp

```diff
@@ -10,19 +11,28 @@
 #include <cstddef>    // for size_t
 #include <cstdint>    // for fixed-width integer types
 #include <cstdio>     // for assertion diagnostics
-#include <cstdlib>    // for abort
 #include <functional> // for less and equal_to
 #include <iterator>   // for reverse_iterator and iterator traits
 #include <limits>     // for numeric_limits
 #include <memory>     // for destroy
 #include <new>        // for operator new
 #include <ranges>
-#include <stdexcept>   // for length_error
 #include <type_traits> // for aligned_storage and all meta-functions
 
 // Artifact from previous implementation, can be used as hints for optimizer
 #define IV_EXPECT(EXPR)
 
+#ifndef BEMAN_IV_THROW_OR_ABORT
+#if !defined(__cpp_exceptions) || __cpp_exceptions < 199711L ||                \
+    BEMAN_INPLACE_VECTOR_NO_EXCEPTIONS()
```

Comment:
> wusatosi do you mean checking it in the config.hpp or in CMake itself using something like check_cxx_source_compiles ? I have pushed an example of how I would imagine it.

Yes I am referring to using `check_cxx_source_compiles` with something like:

```c++
#if !defined(__cpp_exceptions) || __cpp_exceptions < 199711L
#error "No exception Support"
#endif

void main() { throw 0; }
```

And if this does not compile, we default `BEMAN_INPLACE_VECTOR_NO_EXCEPTIONS` to ON with a message.

We then update these lines to:
```suggestion
#if BEMAN_INPLACE_VECTOR_NO_EXCEPTIONS()
```

Reply:
imho checking it inside of the cmake is pretty useless, as it brings to real value over just having the CMake option (whole lib must have exceptions enabled or disabled)

Reply:
If we cant have `__cpp_exceptions` inside of the header, I would rather have only Cmake option, while letting user to override `BEMAN_IV_THROW_OR_ABORT()` if needed

```cpp
#ifndef BEMAN_IV_THROW_OR_ABORT
#if BEMAN_INPLACE_VECTOR_NO_EXCEPTIONS()
#include <cstdlib> // for abort
#define BEMAN_IV_THROW_OR_ABORT(x) abort()
#else
#include <stdexcept> // for length_error
#define BEMAN_IV_THROW_OR_ABORT(x) throw x
#endif
#endif

```

Reply:
I have pushed some changes to avoid NOFLAGFORK roadblock and updated tests accordingly.


I did try replacing `EXPECT_THROW` with `EXPECT_DEATH` when the CMake flag is on, but it extremely increases pipeline runtime...

```
baseline with custom BEMAN_IV_THROW_OR_ABORT override
13 jobs in 8m 5s

add noexcept jobs; replace EXPECT_THROW with EXPECT_DEATH
17 jobs in 21m 49s

add noexcept jobs; replace EXPECT_THROW with do{}while(0); only run noexcept tests with corresponding cmake option
17 jobs in 8m 4s
```